src/Tools/ExternalAccess/Razor/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -37,8 +38,20 @@ internal sealed class RazorAnalyzerAssemblyResolver() : IAnalyzerAssemblyResolve
                         return loadedAssembly;
                     }
 
-                    var assembly = Path.Combine(rootDirectory, $"{assemblyName.Name}.dll");
-                    return compilerContext.LoadFromAssemblyPath(assembly);
+                    // Load all the assemblies, otherwise the compiler ALC could be missing dependencies of the generator.
```

Comment:
Oh ... I see. Because there is no actual handler in the compiler ALC that will call back into this.